kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/DataModelUtil.java

```diff
@@ -101,6 +101,25 @@ public static Schema makeNullableSchema(Schema schema) {
   }
 
   /**
+   * Returns true if the type implements GenericRecord but not SpecificRecord
+   *
+   * @param <E> The entity type
+   * @param type The Java class of the entity type
+   * @return true if the type implements GenericRecord but not SpecificRecord
+   */
+  public static <E> boolean isGeneric(Class<E> type) {
+    // Need to check if SpecificRecord first because specific records also
+    // implement GenericRecord
+    if (SpecificRecord.class.isAssignableFrom(type)) {
+      return false;
+    } else if (IndexedRecord.class.isAssignableFrom(type)) {
```

Comment:
I think this should be GenericRecord instead of IndexedRecord. Indexed doesn't necessarily signal generic. You can also use this to simplify the logic in `createRecord`.